docs/tutorials/Upgrading_to_Application_Commands.md

```diff
@@ -0,0 +1,309 @@
+
+### Restricting command use
+
+[Application command permissions](#DOCS_INTERACTIONS_APPLICATION_COMMANDS/permissions) allow users and apps to restrict command usage in a server. When commands are restricted for a user, they’re greyed out in the client (though in the future they won’t appear at all).
+
+If your app currently relies on permissioning, using command permissions can be a way to port them. It also cleans up the command picker UI for users, making it easier for them to find your app’s commands that are most relevant to them.
+
+### Example payload
+
+Below is a sample payload to create a global slash command with an optional parameter:
+
+```
+{
+    "name": "rock",
+    "type": 1,
+    "description": "Sends a picture of a rock",
+    "options": [
+        {
+            "name": "type",
+            "description": "Type of rock",
+            "type": 3,
+            "required": True,
+            "choices": [
+                {
+                    "name": "Igneous",
+                    "value": "rock_igneous"
+                },
+                {
+                    "name": "Sedimentary",
+                    "value": "rock_sedimentary"
+                },
+                {
+                    "name": "Metamorphic",
+                    "value": "rock_metamorphic"
+                }
+            ]
+        }
+    ]
+ }
+```
+
+#### Default permissions
+
+When [creating](#DOCS_INTERACTIONS_APPLICATION_COMMANDS/create-global-application-command) or [updating](#DOCS_INTERACTIONS_APPLICATION_COMMANDS/edit-global-application-command) a command, you can use the `default_member_permissions` field to restrict the command to a set of bitwise permissions that reflect the default permission flags a user must be granted in order to use the command. 
+
+In addition, the `dm_permission` flag can be used to control whether a global command is available in DMs (not available for guild commands).
+
+#### Updating permissions
+
+Permissions for specific roles, users, and channels can be updated by your app if you have the `applications.commands.permissions.update` scope and a Bearer token that’s authenticated by a user with the necessary user permissions. Details about updating a command’s permissions are [in the documentation](#DOCS_INTERACTIONS_APPLICATION_COMMANDS/permissions).
+
+### After this section
+- Update your app to use the `applications.commands` scope
+- Decide whether each of your commands should be scoped globally (available to all servers) or to a guild (available to specific servers)
+- Decide what permissions your commands may need to run
+- Create your app’s commands using HTTP endpoints
+
+## Handling commands
+
+Commands use the [interactions model](#DOCS_INTERACTIONS_RECEIVING_AND_RESPONDING) rather than WebSocket-based gateways. The interactions model uses HTTP requests to send your app relevant information only when a Discord user triggers one of your app’s interactions.
```

Comment:
```suggestion
Commands use the [interactions model](#DOCS_INTERACTIONS_RECEIVING_AND_RESPONDING) rather than the WebSocket gateway. The interactions model uses HTTP requests to send your app relevant information only when a Discord user triggers one of your app’s interactions.
```